src/jit/gentree.h

```diff
@@ -120,6 +120,8 @@ enum genTreeKinds
     GTK_NOVALUE = 0x0400, // node does not produce a value
     GTK_NOTLIR  = 0x0800, // node is not allowed in LIR
 
+    GTK_NOCONTAIN = 0x1000, // this node is a value, but may not be contained
```

Comment:
> Contained LCL_VAR nodes are always memory

Yes but contained means that no register has been assigned to the node. So it's not clear how lowering could do the contained mem thing before RA runs.

> I can imagine folding GT_IND and contained GT_LCL_VAR into e.g. GT_MEM which would denote a memory access at some address that is described by either a lclVar number or a (possible target-dependent) address mode and obviate the need for GT_LEA in these situations

It's interesting that we may be able to fold `GT_IND` and `GT_LEA` into a single node but the same cannot be done easily with `GT_STOREIND` and `GT_LEA` because the resulting node would have 3 operands. I suspect that will cause some complications due to `GTF_REVERSE_OPS`.